addons/sale_stock/models/sale_order.py

```diff
@@ -585,6 +585,17 @@ def _action_launch_stock_rule(self, previous_product_uom_qty=False):
                 line.name, line.order_id.name, line.order_id.company_id, values))
         if procurements:
             self.env['procurement.group'].run(procurements)
+
+        orders = self.mapped('order_id')
+        for order in orders:
+            pickings_to_assign = order.picking_ids.filtered(
+                lambda p: (p.state == 'confirmed' or (p.state in ['waiting', 'assigned'] and not p.printed))
+                and (p.picking_type_id.reservation_method == 'at_confirm'
+                or (p.picking_type_id.reservation_method == 'by_date' and p.scheduled_date - timedelta(days=p.picking_type_id.reservation_days_before) <= fields.Datetime.now())))
+            if pickings_to_assign:
+                # Trigger the Scheduler for Pickings
+                pickings_to_assign.action_confirm()
+                pickings_to_assign.action_assign()
```

Comment:
This block should not exist anymore. Should be handle in stock.

Reply:
But we will still need the `picking.action_confirm()` to avoid bringing back this bug: https://github.com/odoo/odoo/commit/08af6a3f663abb49d2e77c706e85118a7199ef89 correct?

Reply:
You are right. Keep it for the moment